src/IndexedDB/Browser.IndexedDB.fs

```diff
@@ -121,21 +155,39 @@ type [<AllowNullLiteral; Global>] IDBDatabase =
     abstract objectStoreNames: DOMStringList with get
 
     abstract close: unit -> unit
-    abstract createObjectStore: name: string * ?options: obj -> IDBObjectStore
+    abstract createObjectStore: name: string * ?options: IDBCreateStoreOptions -> IDBObjectStore
     abstract deleteObjectStore: name: string -> unit
-    abstract transaction: storeNames: #seq<string> * ?mode: string * ?options: obj -> IDBTransaction
-    abstract transaction: storeNames: string * ?mode: string * ?options: obj -> IDBTransaction
+    abstract transaction: storeNames: #seq<string> * ?mode: IDBTransactionMode * ?options: IDBTransactionOptions -> IDBTransaction
+    abstract transaction: storeNames: string * ?mode: IDBTransactionMode * ?options: IDBTransactionOptions -> IDBTransaction
 
+    abstract onclose: (Event -> unit) with get, set
+    abstract onversionchange: (Event -> unit) with get, set
+    abstract onabort: (Event -> unit) with get, set
     abstract onerror: (Event -> unit) with get, set
 
 type [<AllowNullLiteral; Global>] IDBOpenDBRequest =
     inherit IDBRequest
 
     abstract blocked: (Event -> unit) with get, set
-    abstract onupgradeneeded: (Event -> unit) with get, set
+    abstract onupgradeneeded: (IDBVersionChangeEvent -> unit) with get, set
 
 type [<AllowNullLiteral; Global>] IDBFactory =
-    abstract ``open``: name: string * ?version: int64 -> IDBOpenDBRequest
+    abstract ``open``: name: string * ?version: int -> IDBOpenDBRequest
     abstract cmp: first: 'T * second: 'T -> int
     abstract deleteDatabase: name: string -> IDBOpenDBRequest
     abstract databases: unit -> JS.Promise<DatabasesType array>
+
+[<AutoOpen>]
+module Static =
+    type IDBKeyRange with
+        [<Emit("IDBKeyRange.bound($0, $1, $2, $3)")>]
+        static member bound(lower: obj, upper: obj, ?lowerOpen: bool, ?upperOpen: bool) = jsNative<IDBKeyRange>
+
+        [<Emit("IDBKeyRange.only($0)")>]
+        static member only(only: obj) = jsNative<IDBKeyRange>
+
+        [<Emit("IDBKeyRange.lowerBound($0, $1)")>]
+        static member lowerBound(lower: obj, ?``open``: bool) = jsNative<IDBKeyRange>
+
+        [<Emit("IDBKeyRange.upperBound($0, $1)")>]
+        static member upperBound(upper: obj, ?``open``: bool) = jsNative<IDBKeyRange>
```

Comment:
Does the code works if we replace this block with:

```fs
type [<AbstractClass; AllowNullLiteral; Global>] IDBKeyRange =
    abstract lower: obj with get
    abstract upper: obj with get
    abstract lowerOpen: bool with get
    abstract upperOpen: bool with get
    abstract includes: obj -> bool

    [<Emit("IDBKeyRange.bound($0, $1, $2, $3)")>]
    static member bound(lower: obj, upper: obj, ?lowerOpen: bool, ?upperOpen: bool) = jsNative<IDBKeyRange>

    [<Emit("IDBKeyRange.only($0)")>]
    static member only(only: obj) = jsNative<IDBKeyRange>

    [<Emit("IDBKeyRange.lowerBound($0, $1)")>]
    static member lowerBound(lower: obj, ?``open``: bool) = jsNative<IDBKeyRange>

    [<Emit("IDBKeyRange.upperBound($0, $1)")>]
    static member upperBound(upper: obj, ?``open``: bool) = jsNative<IDBKeyRange>
```

This is to write more idiomatic F# code.

Reply:
I'll give that a try, should work. Looks like it might be possible to declare them directly [on interfaces](https://github.com/fsharp/fslang-design/blob/main/FSharp-8.0/FS-1128-allow-static-members-in-interfaces.md) soon, which would be handy for stuff like this.